### PR TITLE
packaging/opkg: add upgrade function

### PR DESCRIPTION
**Pull request by aparcar**
##### SUMMARY

introduce the `latest` state which will check if a package is upgradable
and performs an upgrade if available.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

packaging/opkg

##### ADDITIONAL INFORMATION

There have been [such efforts some years ago](https://github.com/ansible/ansible/pull/8751), however they missed to actually check if a package was upgradeable or not.

Also [blacked](https://github.com/ambv/black) the code for better readability, hope that's fine!

### CODE REVIEW

**Comment by skinp** [comment]
@aparcar seems useful. Can you have a look at this test failure?
Also, can we seperate the logic and formatting changes for easier review?

**Comment by aparcar** [comment]
@skinp should work now

**Comment by aparcar** [comment]
Now it's fixed and tested, would be great to merge it :)!

File: lib/ansible/modules/packaging/os/opkg.py [hunk 1]
```diff
@@ -148,12 +157,12 @@ def install_packages(module, opkg_path, packages):
     install_c = 0
 
     for package in packages:
-        if query_package(module, opkg_path, package):
+        if query_package(module, opkg_path, package, latest):
             continue
 
         rc, out, err = module.run_command("%s install %s %s" % (opkg_path, force, package))
```
Comment (skinp) [inline lib/ansible/modules/packaging/os/opkg.py:163]: Does `opkg install` actually upgrade the package if it needs to? [Docs](https://openwrt.org/docs/guide-user/additional-software/opkg) seem to say there's an `upgrade` subcommand?

Comment (aparcar) [reply lib/ansible/modules/packaging/os/opkg.py:163]: I did some testing but the behavior seems to be the same. Both `install` and `upgrade` install the package missing and upgrade the package if an upgrade is available (if the repos where updated previously).

Comment (aparcar) [reply lib/ansible/modules/packaging/os/opkg.py:163]: @dangowrt right?

Comment (skinp) [reply lib/ansible/modules/packaging/os/opkg.py:163]: Alright we'll trust your tests at this point. Thanks!

**Review by skinp — APPROVED** [review]

**Comment by skinp** [comment]
shipit

**Comment by aparcar** [comment]
@skinp this PR is still open, what needs to happen next?

**Pull request closed without merging**